packages/renderer-commonmark/src/index.ts

```diff
@@ -64,34 +107,52 @@ export default class CommonmarkRenderer extends Renderer {
    * ###### and six `#` being the least important
    */
   *'heading'(props: { level: number }): IterableIterator<string> {
-    let heading = yield;
+    let text = yield;
     let level = new Array(props.level + 1).join('#');
-    return `${level} ${heading.join('')}\n\n`;
+    let heading = text.join('');
+
+    // Multiline headings are supported for level 1 and 2
+    if (heading.indexOf('\n') !== -1) {
+      if (props.level === 1) {
+        return `${heading}\n====\n`;
+      } else if (props.level === 2) {
+        return `${heading}\n----\n`;
+      }
+    }
+    return `${level} ${heading}\n`;
   }
 
   /**
    * A horizontal rule separates sections of a story
-   * ---
+   * ***
    * Into multiple sections.
    */
   *'horizontal-rule'(): IterableIterator<string> {
-    return '---\n\n';
+    return '***\n';
   }
 
   /**
    * Images are embedded like links, but with a `!` in front.
    * ![CommonMark](http://commonmark.org/images/markdown-mark.png)
    */
-  *'image'(props: { alt: string, url: string }): IterableIterator<string> {
+  *'image'(props: { alt: string, title?: string, url: string }): IterableIterator<string> {
+    if (props.title) {
+      let title = props.title.replace(/"/g, '\\"');
+      return `![${props.alt}](${props.url} "${title}")`;
+    }
     return `![${props.alt}](${props.url})`;
   }
 
   /**
    * Italic text looks like *this* in Markdown.
    */
-  *'italic'(): IterableIterator<string> {
+  *'italic'(_, state: State): IterableIterator<string> {
+    let isItalicized = state.get('isItalicized');
+    state.set('isItalicized', true);
     let [before, text, after] = yield* split();
-    return `${before}*${text}*${after}`;
+    state.set('isItalicized', isItalicized);
+    let markup = isItalicized ? '_' : '*';
+    return `${before}${markup}${text}${markup}${after}`;
```

Comment:
This is to handle strong italics, which is possible with `*_italic_*` *_italic_*. It's quite absurd to support, but it's part of the spec, so it's here.

`split` should probably be renamed to something else. It returns a triple that pushes any leading or trailing whitespace to the first and last elements.

So, calling `yield* split()` where the `yield` returned `  hello` would return `['  ', 'hello', '']`.

I made the helper to make it easier to handle whitespace rules. This could probably be made easier through a different helper.